### PR TITLE
feat: add unit tests for internal `_debit` function

### PR DESCRIPTION
**Pull request by Le-Caignec**

### CODE REVIEW

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol [hunk 1]
```diff
@@ -398,4 +398,168 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
+
+        _test_debit(iexecLayerZeroBridgeEthereum, address(rlcToken), true);
+    }
+
+    function test_debit_WithoutApproval_SuccessfulBurn() public {
+        _test_debit(iexecLayerZeroBridgeChainX, address(rlcCrosschainToken), false);
+    }
+
+    function _test_debit(IexecLayerZeroBridgeHarness iexecLayerZeroBridge, address tokenAddress, bool approvalRequired)
+        internal
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol:411]: I don’t know, i do that to make it consistent with the rest of the file.

**Review by gfournieriExec — APPROVED** [review]

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol [hunk 2]
```diff
@@ -458,4 +458,158 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
+    }
+
+    function _testFuzz_debit_Amount(IexecLayerZeroBridgeHarness bridge, IERC20 token, uint256 amount) internal {
+        // Fuzz test with different amounts for testing edge case (0 & max RLC supply)
+        uint256 initialBalance = token.balanceOf(user1);
+        uint256 conversionRate = bridge.decimalConversionRate();
+        uint256 expectedMinAmount = (amount / conversionRate) * conversionRate;
+
+        (uint256 amountSentLD, uint256 amountReceivedLD) =
+            bridge.exposed_debit(user1, amount, expectedMinAmount, DEST_EID);
```
Comment (zguesmi) [inline test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol:569]: I don't understand the goal here. Why not?
```suggestion
        uint256 expectedMinAmount = (amount / conversionRate) * conversionRate;

        (uint256 amountSentLD, uint256 amountReceivedLD) =
            bridge.exposed_debit(user1, amount, amount, DEST_EID);
```

**Review by zguesmi — COMMENTED** [review]
We need to test the slippage mechanism. Check that `_debit` reverts with `SlippageExceeded` when the slippage amount is exceeded.

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol [hunk 3]
```diff
@@ -458,4 +458,158 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
+        assertEq(amountReceivedLD, TRANSFER_AMOUNT, "Amount received should equal transfer amount");
+    }
+
+    function test_debit_WithApproval_InsufficientApproval() public {
+        // Setup: User approves less than required
+        vm.prank(user1);
+        rlcToken.approve(address(iexecLayerZeroBridgeEthereum), TRANSFER_AMOUNT - 1);
+
+        // Should revert with insufficient allowance
+        vm.expectRevert();
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol]: fix Commit: 23a1c55724cdaa92c9b92d6f0f69778fe0d8848c

**Review by zguesmi — APPROVED** [review]
